src/renderer/components/ReactFlowBox.tsx

```diff
@@ -135,19 +148,69 @@ const updateZIndexes = (
         }
     }
 
+    const nodesToAdjust: { node: Node<NodeData>; zIndex: number }[] = [];
     // set the zIndex of all edges
     for (const e of edges) {
         let zIndex = Math.max(
             byId.get(e.source)?.zIndex ?? STARTING_Z_INDEX,
             byId.get(e.target)?.zIndex ?? STARTING_Z_INDEX
         );
 
+        const connectedNodes = nodes.filter((n) => n.id === e.source || n.id === e.target);
+
         if (e.selected && zIndex < MIN_SELECTED_INDEX) {
             zIndex += SELECTED_ADD;
+            // If an edge is selected, make the connected nodes the same zIndex
+            connectedNodes.forEach((n) => {
+                if (n.type !== 'iterator') {
+                    // We need to wait until after the loop to adjust the zIndex
+                    nodesToAdjust.push({ node: n, zIndex });
```

Comment:
I cannot see this being correct. Nodes can have multiple edges connected to them, so nodes might end up multiple times in the `nodesToAdjust` array with different zIndex values each.